### DISCUSSION TITLE
Best practice for using it for continuing an app between SSH sessions?

### DISCUSSION THREAD

**Discussion by HubKing** [zellij-org/zellij#3355] [Q&A]
The only thing I am interested is keep an app running when I disconnect/reconnect SSH (e.g, pacman). That is, I don't want multiple sessions or tiling, etc; I just want to easily get back where I was. Under Arch Linux, what is the recommended practice?

I installed it from the "extra" repository, and I thought there may be a service, but there was not. So, far the way I have found is that I need to run `systemd-run --scope --user zellij` when starting it for the first time, and then `zellij attach` from the next time. Typing this every time is tedious, so maybe I could make some sort of script that does something like

```
if `systemd-run --scope --user zellij` is already running
    run `zellij attach`
else
    run `systemd-run --scope --user zellij`
```

and use that script instead of using the command `zellij` directly. But is there any easier, better way?

**Comment by Lockszmith-GH** [comment]
I don't think `zellij` needs `systemd` at all, although I'm not against it.

Below is what I use in my shell `rc` script, which loads upon login:
Depending on whether you are using `bash` or `zsh` as your shell, you can place the following in your `~/.bashrc` or `~/.zshrc`:

```shell
BASE_SHELL=$(basename "$SHELL")
zellij() {
    if [[ -x /tmp/zellij/bootstrap/zellij ]]; then
        /tmp/zellij/bootstrap/zellij "${@}"
    else
        bash <(curl -sL zellij.dev/launch) "${@}"
    fi
}
if [[ -z "$ZELLIJ_SESSION_NAME" ]]; then
    # shellcheck disable=SC2210
    zellij attach -c "$USER@$(hostname)" || EXITCODE=$?

    # shellcheck disable=SC2086 # EXITCODE might be empty
    return $EXITCODE
fi
if [[ -z "$ZSH_VERSION" ]] || (type compdef > /dev/null 2>&1); then
    # shellcheck disable=SC1090
    . <( zellij setup --generate-completion "$BASE_SHELL" ) 2> /dev/null
fi
```

The code always uses the same session name `"$USER@$(hostname)"` - which is a your equivalent of `user@example`. This is the only thing you need to worry about.

`zellij`'s own installer saves a copy of the zellij executable in `/tmp/zellij/bootsrap/zellij`.
This code, checks whether the executable is there and uses it.
If the file isn't there (which will happen afer every reboot automatically), it runs the installer directly from `zellij`'s web site, which will download the executable into the `/tmp/`... location.

The `$ZELLIJ_SESSION_NAME` variable is set while you are running insde a `zellij` session, and so, if it missing, it will execute the 'smart` zellij launch code.

Finally, it runs zellij auto-completion code, making working with zellij much more convinent.

Hope this helps.

This code is based on `zellij`'s own `zellij config

<sup>1</sup> If you're system's /tmp is mounted with `noexec`, you will want to copy it over to a location you _can_ execute from (like `~/bin`).
